src/lib/wallet.ts

```diff
@@ -140,11 +140,11 @@ function generate_keys_bip39(seed = undefined, derivationIndex = 0) {
 		finalSeed = bip39.mnemonicToSeedSync(finalMnemonic).toString("hex");
 	}
 
+	let hdkey = HDKey.fromMasterSeed(finalSeed);
 	const derivationPath = "m/44'/789'/" + derivationIndex + "'/0'/0'";
-	const { key, chainCode } = bip32.derivePath(derivationPath, finalSeed, 0x80000000);
-
-	const privateKey = key.toString("hex");
-	const publicKey = bip32.getPublicKey(key, false).toString("hex");
+	hdkey = hdkey.derive(derivationPath);
+	const privateKey = helpers.buf2hex(hdkey.privateKey);
+	const publicKey = helpers.buf2hex(hdkey.publicKey).slice(2);
 
```

Comment:
`buf2hex` was used for getting the `privateKey` and `publicKey` in hex. But because `buf2hex` was not properly typed, it returned `any`.